File: pythongrind/aoc2024/p3.py
with open("in.txt", "r") as file:
    lines = file.readlines()
cnt = 0
for line in lines:
    arr = list(map(int, line.split()))
    valid = True
    help = 0
    for i in range(len(arr) - 1):
        diff = arr[i+1] - arr[i]
        if diff * help < 0:
            valid = False
        help = diff
        diff = abs(diff)
        if diff < 1 or diff > 3:
            valid = False
    if valid:
        cnt += 1
print(cnt)

File: random/aoc2024/p3.java
import java.io.*;
import java.util.*;

public class p3 {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(new File("in.txt"));
        int cnt = 0;
        while (sc.hasNextLine()) {
            String[] line = sc.nextLine().split(" ");
            ArrayList<Integer> list = new ArrayList<>();
            for (String s : line) {
                list.add(Integer.parseInt(s));
            }
            int prev = 0;
            boolean valid = true;
            for (int i = 0; i < list.size() - 1; i++) {
                int diff = list.get(i + 1) - list.get(i);
                if (prev * diff < 0) { valid = false; }
                prev = diff;
                diff = Math.abs(diff);
                if (diff < 1 || diff > 3) { valid = false; }
            }
            if (valid) { cnt++; }
        }
        System.err.println(cnt);
        sc.close();
    }
}
